crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -51,6 +56,209 @@ use sui_types::sui_serde::{Base64, Encoding};
 mod rpc_types_tests;
 
 pub type GatewayTxSeqNumber = u64;
+pub type SuiMoveTypeParameterIndex = u16;
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct SuiMoveAbilitySet(u8);
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub enum SuiMoveVisibility {
+    Private = 0x0,
+    Public = 0x1,
+    Friend = 0x3,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct SuiMoveStructTypeParameter {
+    pub constraints: SuiMoveAbilitySet,
+    pub is_phantom: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct MoveNormalizedField {
+    pub name: String,
+    pub type_: MoveNormalizedType,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct MoveNormalizedStruct {
+    pub abilities: SuiMoveAbilitySet,
+    pub type_parameters: Vec<SuiMoveStructTypeParameter>,
+    pub fields: Vec<MoveNormalizedField>,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub enum MoveNormalizedType {
+    Bool,
+    U8,
+    U64,
+    U128,
+    Address,
+    Signer,
+    Struct {
+        address: String,
+        module: String,
+        name: String,
+        type_arguments: Vec<MoveNormalizedType>,
+    },
+    Vector(Box<MoveNormalizedType>),
+    TypeParameter(SuiMoveTypeParameterIndex),
+    Reference(Box<MoveNormalizedType>),
+    MutableReference(Box<MoveNormalizedType>),
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct MoveNormalizedFunction {
+    pub visibility: SuiMoveVisibility,
+    pub is_entry: bool,
+    pub type_parameters: Vec<SuiMoveAbilitySet>,
+    pub parameters: Vec<MoveNormalizedType>,
+    pub return_: Vec<MoveNormalizedType>,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct SuiMoveModuleId {
```

Comment:
I noticed that sometimes we prepend `Sui` and sometimes we don't. Let's be consistent about the convention. My preference is to have them start with `Sui` to signal that these are RPC types.